anesthetic/plotting/_matplotlib/boxplot.py

```diff
@@ -0,0 +1,122 @@
+import pandas.plotting._matplotlib.boxplot
+from pandas.plotting._matplotlib.boxplot import BoxPlot as _BoxPlot
+from anesthetic.plotting._matplotlib.core import _WeightedMPLPlot, _get_weights
+from anesthetic.utils import quantile
+from pandas.core.dtypes.missing import remove_na_arraylike
+import numpy as np
+from pandas.io.formats.printing import pprint_thing
+from anesthetic._code_utils import replace_inner_function
+
+
+def _bxpstat(y, weights, whis):
+    q1, med, q3 = quantile(y, [0.25, 0.5, 0.75], weights)
+    bxpstat = {}
+    bxpstat['med'] = med
+    bxpstat['q1'] = q1
+    bxpstat['q3'] = q3
+    bxpstat['whislo'] = q1 - whis*(q3-q1)
+    bxpstat['whishi'] = q3 + whis*(q3-q1)
+    return bxpstat
+
+
+class BoxPlot(_WeightedMPLPlot, _BoxPlot):
+    # noqa: disable=D101
+    @classmethod
+    def _plot(cls, ax, y, column_num=None, return_type="axes", **kwds):
+        if y.ndim == 2:
+            y = [remove_na_arraylike(v) for v in y]
+            # Boxplot fails with empty arrays, so need to add a NaN
+            #   if any cols are empty
+            # GH 8181
+            y = [v if v.size > 0 else np.array([np.nan]) for v in y]
+        else:
+            y = remove_na_arraylike(y)
+
+        weights = kwds.pop("weights", None)
+
+        if weights is None:
+            bp = ax.boxplot(y, **kwds)
+        else:
+            whis = kwds.pop("whis", 1.5)
+            kwds['showfliers'] = False
+            y = np.atleast_2d(y)
+            bp = ax.bxp([_bxpstat(yi, weights, whis) for yi in y], **kwds)
```

Comment:
New code: weighted boxplots using ax.bxp(stats,...) rather than ax.boxplot

Reply:
Yes, `ax.boxplot` doesn't accept weights, so you have to manually compute the weighted boxplots statistics and use the underlying `ax.bxp` which matplotlib's `ax.boxplot` calls underneath.